Source/Core/DolphinQt/GCMemcardManager.h

```diff
@@ -22,6 +24,16 @@ class QPushButton;
 class QTableWidget;
 class QTimer;
 
+struct GCMemcardManagerIconAnimationData
```

Comment:
This can be forward declared in the private section of the class and be defined in the cpp file, as std::vector allows the use of incomplete types. e.g.

```cpp
class GCMemcardManager
{
public:
   // ...

private:
  struct IconAnimationData;

  // ...
};

// cpp file

struct GCMemcardManager::IconAnimationData
{
  // Individual frames
  std::vector<QPixmap> m_frames;

  // Contains a list of frame indices that indicate, for each time unit,
  // the frame that should be displayed when at that time unit
  std::vector<u8> m_frame_timing;
};

// ...
```

Reply:
I'm surprised this is valid even though it's a return value of a member function, but sure, that's certainly cleaner.

Reply:
A return value is allowed to be forward declared, so this is indeed perfectly valid.